File: ruby/reverse_string.rb
def reverse_string(str)
  reverse = ''
  i = str.length - 1
  until (i == -1) do
    reverse += str[i]
    i -= 1
  end
  reverse
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts 'Expecting: "cram"'
  puts "=>", reverse_string('marc')

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: javascript/reverse_string.js
function reverseString(str) {
  let newStr = ''
  for (let i=str.length-1; i>=0; i--) {
    newStr += str[i]
  }
  return newStr
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("Expecting: 'kcuf'");
  console.log("=>", reverseString("fuck"));

  console.log("Expecting: 'cram'");
  console.log("=>", reverseString("marc"));

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution
